tests/vibe.web.rest.1140/source/app.d

```diff
@@ -0,0 +1,52 @@
+import std.datetime;
+import vibe.d;
+
+interface IOrientDBRoot
+{
+	@property IOrientDBQuery query();
+}
+
+interface IOrientDBQuery
+{
+	@method(HTTPMethod.GET)
+	@path(":db_name/sql/:query/:result_set_size")
+	Json sql(string _db_name, string _query, int _result_set_size);
+}
+
+class OrientDBRoot : IOrientDBRoot {
+	private OrientDBQuery m_query;
+	override @property IOrientDBQuery query() { return m_query; }
+	public this() { this.m_query = new OrientDBQuery(); }
+}
+
+class OrientDBQuery : IOrientDBQuery {
+	override Json sql(string _db_name, string _query, int _result_set_size) {
+		assert(_db_name == Param1, _db_name);
+		assert(_query == Param2, _query);
+		assert(_result_set_size == Param3, to!string(_result_set_size));
+		return Json.emptyObject;
+	}
+}
+
+enum Param1 = "twitter_data";
+enum Param2 = "select DownloadedDateTime from Message";
```

Comment:
Ok. Will return to this issue tomorrow:)

>In addition, removing this assert is a Bad Idea, since it wouldn't catch other bugs.

We should answer to the simple question: Can Path part contain '/' character? (Not relevant with REST). 
For example, I got create a file with name '\' on linux:
```
[wazar@localhost Proj]$ echo "hello" | cat > '\'
[wazar@localhost Proj]$ cat '\'
hello
```
This example already breaks your assert.
Of course, URL  path part may contain '/' character and it should be encoded as '%2f' (similar to ' ', '?', ':' which also may be in URL part and which should be encoded):
https://specs.webplatform.org/url/webspecs/develop/#path-segment
>A path segment must be zero or more URL units, __excluding "/" and "?"__.
>The URL units are __URL code points__ and __percent-encoded bytes__.
>The URL code points are ASCII alphanumeric, "!", "$", "&", "'", "(", ")", "*", "+", ",", "-", ".", _"/"_, ":", ";", "=", _"?"_
>A percent-encoded byte is "%", followed by two ASCII hex digits. Sequences of percent-encoded bytes, after conversion to bytes, should not cause a utf-8 decoder to run into any errors.

Ergo: URL path segment can't contain '/'directly, however it may contain '%2f' part.
As long as `PathEntry` contains non-encoded path segmants, it may contain a '/' or '?' characters, given that they will be encoded inside `URL` struct.